### ISSUE TITLE
Astro file containing only script will not execute script

### ISSUE THREAD

**Issue by pReya** [withastro/astro#6384]
### What version of `astro` are you using?

2.0.16

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

npm

### What operating system are you using?

Mac

### Describe the Bug

I remember seeing a ticket for this a couple of months back (for Astro 1), but I can't find it anymore. I assume this is a regression:

When creating a `.astro` file that ONLY contains a `<script>`, the script will not be executed in the browser.

As soon as you add any HTML tags to the file (e.g. a `<h1>`), the script will work again.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-vzr332?file=src/pages/index.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by matthewp** [comment]
While technically a bug, this is something we'll likely never fix both because it's a hard problem and because it's not a realistic scenario. In 3.0 we might even make it so that we don't inject scripts unless there is a `<head>` present. Adding a head does fix this scenario too, see https://stackblitz.com/edit/github-vzr332-adnv5h?file=src%2Fpages%2Findex.astro

**Comment by pReya** [comment]
> While technically a bug, this is something we'll likely never fix both because it's a hard problem and because it's not a realistic scenario. In 3.0 we might even make it so that we don't inject scripts unless there is a `<head>` present. Adding a head does fix this scenario too, see [stackblitz.com/edit/github-vzr332-adnv5h?file=src%2Fpages%2Findex.astro](https://stackblitz.com/edit/github-vzr332-adnv5h?file=src%2Fpages%2Findex.astro)

Well, it used to work in v1.9 – so I assumed it to still work in 2.0 when I upgraded. Finding this took me a while. It should at least be properly documented if this won't be fixed.

Also, I disagree on this not being a realistic scenario. This is from an actual project of mine, where I use a `.astro` file for redirect, which looks something like this:

```
<script>
  import { getBaseUrl } from "@utils";
  const baseUrl = getBaseUrl(true);
  const preferredLanguage = LanguageUtils.getOrGuessInitialLanguageFromBrowser();
  window.location.pathname = `${baseUrl}/${preferredLanguage.toLowerCase()}`;
</script>
```

**Comment by rishi-raj-jain** [comment]
@pReya  head script redirection would still required head tag to be present, so something like 

```html
<html>
<head>
<script>
 // Your code
</script>
</head>
</html>
```

shall solve your use case.

**Comment by ematipico** [comment]
> Well, it used to work in v1.9 – so I assumed it to still work in 2.0 when I upgraded. Finding this took me a while. It should at least be properly documented if this won't be fixed

Following the discussion, it seems you were taking advantage of a bug - or a feature -  that was solved when 2.0 shipped, which will be more strict on 3.0.

Considering that there's a workaround to make the script work, I think we can close the issue.
